clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -484,6 +548,25 @@ public void run() {
         return Collections.unmodifiableMap(childrenSensors);
     }
 
+    public MetricName metricInstance(MetricNameTemplate template, String... keyValue) {
```

Comment:
nit: wondering if we can still name these functions as `metricName(..)` or is this intentionally to differentiate with the other functions?